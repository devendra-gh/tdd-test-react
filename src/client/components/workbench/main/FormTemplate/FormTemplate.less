.formTemplate {
  p {
    // margin-bottom: 0;
    line-height: 2.4rem;
    color: #3f3e45;
  }

  .ui-lib-form__fieldset {
    margin-top: 2rem;
    margin-bottom: 2rem !important;
  }

  .ui-lib-form__fieldset-header-wrap {
    margin-bottom: 2rem;
  }
}
