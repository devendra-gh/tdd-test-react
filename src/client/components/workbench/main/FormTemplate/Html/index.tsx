import React from 'react';

import { noop, isFunction } from 'lodash';

import Checkbox from '@tamm/ui-lib-v2-checkbox';
import CheckboxGroup from '@tamm/ui-lib-v2-checkbox-group';
import DatePicker from '@tamm/ui-lib-v2-date-picker';
import Input from '@tamm/ui-lib-v2-input';
import InputNumber from '@tamm/ui-lib-v2-input-number';
import InputPassword from '@tamm/ui-lib-v2-input-password';
import InputTelephone from '@tamm/ui-lib-v2-input-telephone';
import RadioGroup from '@tamm/ui-lib-v2-radio-group';
import RangeInput from '@tamm/ui-lib-v2-range-input';
import DlsButton from '@tamm/ui-lib-v2-button';
import DlsForm from '@tamm/ui-lib-v2-form';
import Select from '@tamm/ui-lib-v2-select';
import Slider from '@tamm/ui-lib-v2-slider';
import TextArea from '@tamm/ui-lib-v2-text-area';
import TimePicker from '@tamm/ui-lib-v2-time-picker';
import Grid from '@tamm/ui-lib-v2-grid';
import { IVariables } from '@tamm/app-composer';

import FileUpload from '../../FileUpload';
import { FieldErrorType } from '../Types';

const CheckboxInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <>
      <Checkbox
        {...field.props}
        onChange={handleInputChange(field)}
        checked={fieldValue}
        validateStatus={fieldError ? 'error' : undefined}
        disabled={disabled}
      />
    </>
  );
};

const CheckboxGroupInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <>
      <CheckboxGroup
        {...field.props}
        onChange={handleInputChange(field)}
        i18n={noop}
        disabled={disabled}
      />
    </>
  );
};

const DatePickerInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <DatePicker
      {...field.props}
      i18n={noop}
      onChange={handleInputChange(field)}
      validateStatus={fieldError ? 'error' : undefined}
      help={
        (fieldError || [])
          .map((error: FieldErrorType) => error.message)
          .join('. ') || field.props.help
      }
      disabled={disabled}
      value={fieldValue}
    />
  );
};

const TextInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <>
      <Input
        {...field.props}
        onChange={handleInputChange(field)}
        value={fieldValue}
        validateStatus={fieldError ? 'error' : undefined}
        disabled={disabled}
        help={
          (fieldError || [])
            .map((error: FieldErrorType) => error.message)
            .join('. ') || field.props.help
        }
      />
    </>
  );
};

const NumberInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <>
      <InputNumber
        {...field.props}
        onChange={handleInputChange(field)}
        value={fieldValue}
        validateStatus={fieldError ? 'error' : undefined}
        disabled={disabled}
        help={
          (fieldError || [])
            .map((error: FieldErrorType) => error.message)
            .join('. ') || field.props.help
        }
      />
    </>
  );
};

const PasswordInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <>
      <InputPassword
        {...field.props}
        onChange={handleInputChange(field)}
        value={fieldValue}
        validateStatus={fieldError ? 'error' : undefined}
        disabled={disabled}
        help={
          (fieldError || [])
            .map((error: FieldErrorType) => error.message)
            .join('. ') || field.props.help
        }
      />
    </>
  );
};

const TelephoneInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <>
      <InputTelephone
        {...field.props}
        onSelect={handleInputChange(field)}
        value={fieldValue}
        validateStatus={fieldError ? 'error' : undefined}
        disabled={disabled}
        help={
          (fieldError || [])
            .map((error: FieldErrorType) => error.message)
            .join('. ') || field.props.help
        }
      />
    </>
  );
};

const TimePickerInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <>
      <TimePicker
        {...field.props}
        value={fieldValue}
        onChange={handleInputChange(field)}
        validateStatus={fieldError ? 'error' : undefined}
        help={
          (fieldError || [])
            .map((error: FieldErrorType) => error.message)
            .join('. ') || field.props.help
        }
        disabled={disabled}
      />
    </>
  );
};

const RadioInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  const items = (field.props.items || []).map((item: IVariables) => {
    if (fieldValue) {
      if (item.label === fieldValue) return { ...item, checked: true };
      return { ...item, checked: false };
    }
    return item;
  });
  return (
    <RadioGroup
      {...field.props}
      items={items}
      onChange={handleInputChange(field)}
      value={fieldValue}
      validateStatus={fieldError ? 'error' : undefined}
      disabled={disabled}
      help={
        (fieldError || [])
          .map((error: FieldErrorType) => error.message)
          .join('. ') || field.props.help
      }
    />
  );
};

const InputRange = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <RangeInput
      {...field.props}
      onChange={handleInputChange(field)}
      value={fieldValue}
      validateStatus={fieldError ? 'error' : undefined}
      disabled={disabled}
      help={
        (fieldError || [])
          .map((error: FieldErrorType) => error.message)
          .join('. ') || field.props.help
      }
    />
  );
};

const SelectInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;

  return (
    <Select
      {...field.props}
      onChange={handleInputChange(field)}
      value={fieldValue}
      multi={field.multi || false}
      disabled={disabled}
      validateStatus={fieldError ? 'error' : undefined}
      help={
        (fieldError || [])
          .map((error: FieldErrorType) => error.message)
          .join('. ') || field.props.help
      }
    />
  );
};

const TextAreaInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <TextArea
      {...field.props}
      onChange={handleInputChange(field)}
      value={fieldValue}
      disabled={disabled}
      validateStatus={fieldError ? 'error' : undefined}
      help={
        (fieldError || [])
          .map((error: FieldErrorType) => error.message)
          .join('. ') || field.props.help
      }
    />
  );
};

const FileInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <FileUpload
      {...field.props}
      i18n={noop}
      disabled={disabled}
      validateStatus={fieldError ? 'error' : undefined}
      validationMessage={
        (fieldError || [])
          .map((error: FieldErrorType) => error.message)
          .join('. ') || field.props.validationMessage
      }
      files={fieldValue || []}
      onChange={handleInputChange(field)}
      onRemove={handleInputChange(field)}
    />
  );
};

const SliderInput = ({
  field,
  fieldValue,
  fieldError,
  handleInputChange,
}: any) => {
  const { disabled } = field;
  return (
    <>
      <Slider
        {...field.props}
        onChange={handleInputChange(field)}
        value={fieldValue}
        validateStatus={fieldError ? 'error' : undefined}
        disabled={disabled}
        help={
          (fieldError || [])
            .map((error: FieldErrorType) => error.message)
            .join('. ') || field.props.help
        }
      />
    </>
  );
};

const Form = ({
  children,
  handleCancel,
  handleSubmit,
  handleBack,
  onSubmitLabel,
  onBackLabel,
  title,
  props,
  includeCancelButton,
  includeBackButton,
  btnSubmitArrow,
  btnBackArrow,
  formValid,
  defaultSubmitLabel,
  defaultBackLabel,
  defaultCancelLabel,
}: any) => {
  return (
    <div>
      <DlsForm
        backButton={
          includeBackButton
            ? {
                label: onBackLabel || defaultBackLabel,
                onClick: (e: React.FormEvent<HTMLFormElement>) => {
                  // eslint-disable-next-line no-unused-expressions
                  handleBack && isFunction(handleBack) && handleBack(e, props);
                },
                withArrow: btnBackArrow,
                alignIcon: btnBackArrow,
              }
            : undefined
        }
        submitButton={{
          label: onSubmitLabel || defaultSubmitLabel,
          onClick: (e: React.FormEvent<HTMLFormElement>) => {
            handleSubmit(e, props);
          },
          withArrow: btnSubmitArrow,
          alignIcon: btnSubmitArrow,
          disabled: !formValid,
        }}
        cancelLink={
          includeCancelButton
            ? {
                label: defaultCancelLabel,
                href: '',
                onClick: (e: React.FormEvent<HTMLFormElement>) => {
                  // eslint-disable-next-line no-unused-expressions
                  handleCancel &&
                    isFunction(handleCancel) &&
                    handleCancel(e, props);
                },
              }
            : undefined
        }
      >
        {children}
      </DlsForm>
    </div>
  );
};

const FieldGroup = ({ field, children, width }: any) => {
  return <div key={field.name}>{children}</div>;
};

const SectionForm = ({ title, children }: any) => {
  return null;
};

const Button = ({ field, handleInputChange }: any) => {
  return (
    <DlsButton {...field.props} aria-label="button-primary" size="small" />
  );
};

export {
  Button,
  CheckboxInput,
  CheckboxGroupInput,
  DatePickerInput,
  FieldGroup,
  FileInput,
  Form,
  Grid,
  InputRange,
  NumberInput,
  PasswordInput,
  RadioInput,
  SelectInput,
  SectionForm,
  TextInput,
  TelephoneInput,
  TextAreaInput,
  TimePickerInput,
  SliderInput,
};
