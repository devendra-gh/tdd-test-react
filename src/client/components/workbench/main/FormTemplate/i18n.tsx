const errorMessages: any = {
  'form.error.required': 'This field is required',
  'form.error.invalid': 'This field is invalid',
  'form.error.email': 'This should be email',
  'form.error.number': 'This should be numbers',
  'form.error.emiratesId': 'form.error.emiratesId',
  'form.error.phone': 'This should be phone',
  'form.error.alpha': 'form.error.alpha',
  'form.error.digits': 'form.error.digitsd',
  'form.error.fileMaxSize': 'form.error.fileMaxSize',
  'form.error.file': 'form.error.file',
  'form.error.invalidFileType': 'form.error.invalidFileTyped',
};

export const i18n = (text: string) => errorMessages[text];
