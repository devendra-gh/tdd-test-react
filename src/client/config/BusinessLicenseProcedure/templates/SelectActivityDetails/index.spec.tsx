import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from 'react-router';
import { IVariables } from '@tamm/app-composer';
import { Viewport, ViewportProvider } from '@tamm/ui-lib-v2-viewport';
import FormTemplate from './index';

jest.mock('./Components/CategoryInput/index', () => () => 'CategoryInput');
jest.mock('./Components/ActivityTable/index', () => () => 'ActivityTable');
jest.mock('./Components/SearchInput/index', () => () => 'SearchInput');
jest.mock('./Components/AlertComponent/index', () => () => 'AlertComponent');
jest.mock('./Components/PaginationActivity/index', () => () =>
  'PaginationActivity',
);

jest.mock('@tamm/app-composer', () => ({
  withTemplateHooks: (component: any) => component,
  noticeTypes: {
    INFO: 'INFO',
    SUCCESS: 'SUCCESS',
    WARNING: 'WARNING',
  },
}));

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('FormTemplate', () => {
  let props: IVariables;
  window.scrollTo = jest.fn();

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation(query => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });

    props = {
      history: {
        push: jest.fn(),
      },
      actions: {
        stepsStatus: {
          update: jest.fn(),
        },
        formSelectActivity: {
          update: jest.fn(),
        },
      },
      getInitialState: jest.fn(),
      formSelectActivity: {
        category: 'category',
        subCategory: 'subCategory',
        inputType: 'CATEGORY',
        loading: true,
      },
      onChange: jest.fn(),
      onSubmit: jest.fn(),
      i18n: (key: string) => key,
      title: 'FormTemplate',
      subTitle: 'subTitle',
      description: 'desc',
      locale: 'en',
      currentStep: 'economic_licence',
      currentSubStep: 'payment',
      buttons: [
        {
          label: 'btn',
          onClick: jest.fn(),
        },
      ],
      steps: [],
      stepStatus: {
        economic_name: 'economic_name',
        'economic_name.payment': 'economic_name.payment',
      },
      list: {
        labelHeading: 'label',
        valueHeading: 'value',
        listDetails: {
          label: 'label',
          value: 'value',
        },
      },
      totalSection: 1000,
    };
  });

  it('should render with all props', async () => {
    const { getByText, container } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <FormTemplate {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );

    expect(getByText(props.subTitle)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('submit button click, alert shown', async () => {
    const { getAllByLabelText } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <FormTemplate {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );

    fireEvent.click(getAllByLabelText('button')[1]);
    expect(
      props.actions.formSelectActivity.update.mock.calls[0][0].showError,
    ).toBe(true);
  });

  it('submit button click, props.onSubmit called', async () => {
    props.formSelectActivity.activity = 'activity';
    const { getAllByLabelText } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <FormTemplate {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );

    fireEvent.click(getAllByLabelText('button')[1]);
    expect(props.onSubmit).toBeCalled();
  });

  it('back button click, redirect', async () => {
    props.formSelectActivity.activity = 'activity';
    const { getAllByLabelText } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <FormTemplate {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );

    fireEvent.click(getAllByLabelText('button')[0]);
    expect(props.history.push).toBeCalled();
    // expect(props.actions.stepsStatus.update).toBeCalled();
  });

  it('radio change, onchange fired', async () => {
    props.formSelectActivity.activity = 'activity';
    const { container } = render(
      <MemoryRouter>
        <ViewportProvider>
          <Viewport sm md lg xl>
            <FormTemplate {...props} />
          </Viewport>
        </ViewportProvider>
      </MemoryRouter>,
    );

    fireEvent.click(container.querySelectorAll('.ui-lib-radio__input')[1]);
    expect(props.onChange).toBeCalled();
  });
});
