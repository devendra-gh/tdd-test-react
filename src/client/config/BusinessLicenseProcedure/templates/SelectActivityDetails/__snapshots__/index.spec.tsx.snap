// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTemplate should render with all props 1`] = `
<div
  class="ui-lib-viewport ui-lib-viewport_lg"
>
  <div
    class="container"
  >
    <div
      class="ant-row ui-lib-grid-row ui-lib-grid-row "
      style="margin-left: -10px; margin-right: -10px;"
    >
      <div
        class="ant-col ui-lib-grid-col ui-lib-grid-col-0 ui-lib-grid-col  ui-lib-grid-col-xs-24 ui-lib-grid-col-sm-24 ui-lib-grid-col-md-24 ui-lib-grid-col-lg-16 ui-lib-grid-col-xl-18"
        style="padding-left: 10px; padding-right: 10px;"
      >
        <div
          class="head-section"
        >
          <h3
            class="head-section__title"
          >
            subTitle
          </h3>
          <p
            class="head-section__description"
          >
            desc
          </p>
        </div>
        <div
          class="radio-container"
        >
          <div
            class="ui-lib-radio-group ui-lib-radio-group_horizontal-align"
          >
            
            <div
              class="ui-lib-radio-group__items"
            >
              <label
                class="ui-lib-radio ui-lib-radio_h5"
                for="uil_radio_button.1"
              >
                <input
                  checked=""
                  class="ui-lib-radio__input"
                  data-index="0"
                  id="uil_radio_button.1"
                  name=""
                  type="radio"
                  value="CATEGORY"
                />
                <div
                  class="ui-lib-radio__box ui-lib-radio__box_checked"
                >
                  <div
                    class="ui-lib-radio__box-dot"
                  />
                </div>
                <div>
                  <h5
                    class="ui-lib-radio-label ui-lib-radio-label_h5"
                  >
                    label.businessActivity
                  </h5>
                  
                </div>
              </label>
              <label
                class="ui-lib-radio ui-lib-radio_h5"
                for="uil_radio_button.2"
              >
                <input
                  class="ui-lib-radio__input"
                  data-index="1"
                  id="uil_radio_button.2"
                  name=""
                  type="radio"
                  value="NAME"
                />
                <div
                  class="ui-lib-radio__box"
                >
                  <div
                    class="ui-lib-radio__box-dot"
                  />
                </div>
                <div>
                  <h5
                    class="ui-lib-radio-label ui-lib-radio-label_h5"
                  >
                    label.activityName
                  </h5>
                  
                </div>
              </label>
            </div>
          </div>
        </div>
        <form
          class="ui-lib-form qa-form"
          id=""
        >
          
          <div>
            SearchInput
            CategoryInput
            <div
              style="height: 40px;"
            />
            <div
              class="spinner-wrapper"
            >
              <div
                class="ui-lib-spinner qa-spinner ui-lib-spinner_logo"
              />
            </div>
            AlertComponent
            ActivityTable
            PaginationActivity
          </div>
          <div
            class="ui-lib-form__controls"
          >
            <div
              class="ui-lib-form__controls-buttons"
            >
              <button
                aria-label="button"
                class="ui-lib-button qa-button ui-lib-button_secondary ui-lib-button_default ui-lib-button_icon-start ui-lib-button_space-between ui-lib-button_with-icon"
                type="button"
              >
                button.back
                <svg
                  class="ui-lib-button__icon ui-lib-icon"
                  data-key=""
                  height="24"
                  version="1.1"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title />
                  <path
                    d="M22.548,10.561H5.437a.251.251,0,0,1-.165-.438l8.637-7.6a1.44,1.44,0,0,0-1.9-2.162L.828,10.2a2.4,2.4,0,0,0,0,3.6l11.179,9.837a1.44,1.44,0,0,0,1.9-2.161l-8.637-7.6a.251.251,0,0,1,.165-.438H22.548a1.44,1.44,0,0,0,0-2.879Z"
                  />
                </svg>
              </button>
              <button
                aria-label="button"
                class="ui-lib-button qa-button ui-lib-button_primary ui-lib-button_default ui-lib-button_icon-end ui-lib-button_space-between ui-lib-button_with-icon"
                type="button"
              >
                button.next
                <svg
                  class="ui-lib-button__icon ui-lib-icon"
                  data-key=""
                  height="24"
                  version="1.1"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <title />
                  <path
                    d="M23.987,12a2.411,2.411,0,0,0-.814-1.8L11.994.361a1.44,1.44,0,0,0-1.9,2.162l8.637,7.6a.25.25,0,0,1-.165.437H1.452a1.44,1.44,0,0,0,0,2.88H18.563a.251.251,0,0,1,.165.438l-8.637,7.6a1.44,1.44,0,1,0,1.9,2.161L23.172,13.8A2.409,2.409,0,0,0,23.987,12Z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </form>
        <div
          style="height: 60px;"
        />
      </div>
      <div
        class="ant-col ui-lib-grid-col ui-lib-grid-col-0 ui-lib-grid-col  ui-lib-grid-col-xs-24 ui-lib-grid-col-sm-24 ui-lib-grid-col-md-24 ui-lib-grid-col-lg-8 ui-lib-grid-col-xl-6"
        style="padding-left: 10px; padding-right: 10px;"
      >
        <div
          class="sidebar"
        >
          <h6
            style="font-size: 20px; font-weight: normal; margin-bottom: 5px; color: rgb(22, 16, 56);"
          >
            aspectOfLife
          </h6>
          <h4
            style="margin-bottom: 6rem; font-size: 24px; color: rgb(22, 16, 56);"
          >
            manage_business
          </h4>
          <div
            class="ui-lib-relevant-entity"
          >
            <h5
              class="ui-lib-relevant-entity__title"
            >
              relevant_entity
            </h5>
            <img
              alt="relevant_entity logo"
              class="ui-lib-relevant-entity__logo"
              src="/images/DED-logo.png"
            />
            <div
              class="ui-lib-relevant-entity__sub-title"
            >
              departmentOfEconomicDevelopment
            </div>
            <div
              class="ui-lib-relevant-entity__address"
            >
              departmentOfEconomicDevelopment.address
            </div>
            <div
              class="ui-lib-relevant-entity__labeled-area"
            >
              <div
                class="ui-lib-label"
              >
                phone
              </div>
              <div
                class="ui-lib-relevant-entity__labeled-area-value ui-lib-relevant-entity__labeled-area-value_ltr"
                x-apple-data-detectors="false"
              >
                +971 2 815 8888
              </div>
            </div>
            <div
              class="ui-lib-relevant-entity__labeled-area"
            >
              <div
                class="ui-lib-label"
              >
                email
              </div>
              <div
                class="ui-lib-relevant-entity__labeled-area-value ui-lib-relevant-entity__labeled-area-value_underlined"
                x-apple-data-detectors="false"
              >
                <a
                  class="ui-lib-relevant-entity__link"
                  href="mailto:info@adeconomy.ae"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  info@adeconomy.ae
                </a>
              </div>
            </div>
            <div
              class="ui-lib-relevant-entity__labeled-area"
            >
              <div
                class="ui-lib-label"
              >
                website
              </div>
              <div
                class="ui-lib-relevant-entity__labeled-area-value ui-lib-relevant-entity__labeled-area-value_underlined"
                x-apple-data-detectors="false"
              >
                <a
                  class="ui-lib-relevant-entity__link"
                  href="https://added.gov.ae"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  https://added.gov.ae
                </a>
              </div>
            </div>
            <div
              class="ui-lib-relevant-entity__labeled-area"
            >
              <div
                class="ui-lib-label"
              >
                office hours
              </div>
              <div
                class="ui-lib-relevant-entity__labeled-area-value"
                x-apple-data-detectors="false"
              >
                departmentOfEconomicDevelopment.workingTime
              </div>
            </div>
            
          </div>
          <div
            class="hidden-md"
            style="height: 60px;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
